apps/palette-generator/paletteJsonGenerator.ts

```diff
@@ -0,0 +1,241 @@
+};
+
+// End of configurable options
+
+let bliss_gloss;
+export async function fetchBlissGlossJson () {
+  // Read and parse the Bliss gloss JSON file
+  try {
+    const fetchResponse = await fetch("http://localhost:5173/data/bliss_symbol_explanations.json");
+    bliss_gloss = await fetchResponse.json();
+  } catch (error) {
+    console.error(`Error fetching 'bliss_symbol_explanations.json': ${error.message}`);
+  }
+  return bliss_gloss;
+}
+
+/**
+ * Test for the presencs of a string that encodes SVG builder information.  Such
+ * strings begin with "SVG:" and ends with ":SVG"
+ * @param {string} - the string to test.
+ * @returns {boolean}
+ */
+function isSvgBuilderString (theString) {
+  return theString.startsWith(SVG_PREFIX) && theString.endsWith(SVG_SUFFIX);
+}
+
+/**
+ * Converts a string that encodes the information required by the SvgUtils
+ * (svg builder) to the proper format -- an array of bliss-svg specifications.
+ * @param {string} svgBuilderString - The string to convert.
+ * @return {Array} - An array of the specifiers required by the SvgUtils.
+ * @throws {Error} - If the encoding is not well formed.
+ */
+function convertSvgBuilderString (theString) {
+  // Replace the SVG prefix and suffix strings with square brackers (array)
+  theString = theString.replace(SVG_PREFIX, "[").replace(SVG_SUFFIX,"]");
+  return JSON.parse(theString);
+}
+
+/**
+ * Finds the BCI AV ID(s) for a given label.  The label is compared to each of
+ * the glosses where a match is defined as either an exact match, or a "word"
+ * match using the regular expression /\bword\b/, where "word" is the value of
+ * the given label.
+ * @param {string} label - The label to use to search for matches in the gloss.
+ * @param {Array} blissGlosses - Array of objects containing BCI AV IDs, and
+ *                               their glosses:
+ *                               { id: number, description: string, ... }
+ * @returns {Array} An array of objects whose gloss matches the given label:
+ *                  { id: {number}, description: {string}, ... }
+ * @throws {Error} If no BCI AV ID is found for the label.
+ */
+function findBciAvId(label, blissGlosses) {
+  // Check if the label has a special encoding
+  if (label in specialEncodings) {
+    return specialEncodings[label];
+  }
+  const matches = [];
+  // Search for the label in the Bliss gloss
+  console.log(`For label ${label}:`);
+  for (const gloss of blissGlosses) {
+    // Try an exact match or a word match
+    const wordMatch = new RegExp("\\b" + `${label}` + "\\b");
+    if ((label === gloss.description) || wordMatch.test(gloss.description)) {
+      matches.push({ bciAvId: parseInt(gloss.id), label: gloss.description });
+      console.log(`\tFound match: ${gloss.description}, bci-av-id: ${gloss.id}`);
+    }
+  }
+  // If no BCI AV ID is found, throw an error
+  if (matches.length === 0) {
+    throw new Error(`BciAvId not found for label: ${label}`);
+  }
+  console.log("");
```

Comment:
Shall this line be removed?

Reply:
It was meant to output a blank line to separate the matches logged on the console.  But in FF it insists on printing `<empty string>` which is not a blank line.  I switched it to `console.log("\n");` and that does print a blank line.  Then I took it out completely, and the logged matches are readable even without a separating blank line.  So, no it's been removed.